models/actions/run.go

```diff
@@ -164,6 +164,60 @@ func updateRepoRunsNumbers(ctx context.Context, repo *repo_model.Repository) err
 	return err
 }
 
+// CancelRunningJobs cancels all running jobs of a run
+func CancelRunningJobs(ctx context.Context, run *ActionRun) error {
+	runs, _, err := FindRuns(ctx, FindRunOptions{
+		RepoID: run.RepoID,
+		Ref:    run.Ref,
+		Status: []Status{StatusRunning, StatusWaiting},
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(runs) == 0 {
+		return nil
+	}
+
+	for _, run := range runs {
+		run.Status = StatusCancelled
+		run.Stopped = timeutil.TimeStampNow()
+		if err := UpdateRun(ctx, run, []string{"status", "stopped"}...); err != nil {
+			return err
+		}
```

Comment:
I don't think we need this, `UpdateRunJob` will aggregate and update the status of its `run` automatically.

So we can just `UpdateRunJob` or `StopTask` (it calls `UpdateRunJob`).